### DISCUSSION TITLE
Waveshare RS485 TO ETH (B) Setup Guide

### DISCUSSION THREAD

**Discussion by Ndrinta** [nathanmarlor/foxess_modbus#123] [General]
This is a little quick start guide on how to set-up the [Waveshare RS485 TO ETH (B)](https://www.waveshare.com/wiki/RS485_TO_ETH_(B)).
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230366715-6557b754-9906-4a53-a716-0d0bcbfad8ad.png" width=40% height=40%>
</p>
You can set-up your device from the web interface (by simply browsing to the IP_ADDRESS of the device) or use the provided tool (recommended).

I'll explain how to set it up using the dedicated tool.
To do so, download VirCom_en.rar from [here](https://mega.nz/folder/Xd0TxbZA#ghhjmBHi8gJtsFgY4AkL8w) (make sure that the device is hooked up to you home network first), extract it and run the executable.

You'll be presented with this interface -> Click on the "Device" tab.
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230362254-8221407c-390b-45cb-b0a2-3598fbda6f90.png" width=100% height=100%>
</p>

A new window will appear, you'll see your new device listed - Double click on this line.
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230362469-ea9a39bb-3b25-4192-8372-e7702635055d.png" width=100% height=100%>
</p>

And finally you are presented with the "Device Settings" window.

OPTIONAL: Right out of the bat, you can consider upgrading the firmware to the latest available at the moment (V1.486 - April 2023). If you so desire, check the appendix at end of this guide.

Now, copy the following settings 
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230365611-5fb44e45-eb85-4ee5-95cd-65766c7c1d80.png" width=100% height=100%>
</p>

Of course, ignore the IP_ADDRESS I used, you'll have your own.
Then click on "More Advanced Settings..." and enable the "Enable RS485 bus conflict detection, set 20ms (or something you're comfortable with).

OPTIONAL: You can also enable the Multi-Host setting, not required for a simple single-inverter installation.
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230366542-2ee78ea6-aa1b-4399-b6e6-665a1e3709f7.png" width=100% height=100%>
</p>

Hit the "OK" button (the Advanced Setting window will close), and click on the "Modify Settings" button on the "Device Settings" window.
Then click on "Restart Dev" and wait for it to become available again, connect to it and verify that all settings have been accepted.

OPTIONAL: Click the "Save As Default" button to make these settings the default ones, this should help you to reconfigure the device in case it will loose its settings (by clicking "Load Default").
It's unclear to me where this settings are stored, so more test is needed to understand this feature completely.

---  Appendix ---

- Firmware Upgrade:

From the "Device Settings" window, click on the Firmware/Config button (at the bottom).
NOTE: You will find these same instructions on page 48-52 of the user manual you can find online (or [here](https://mega.nz/folder/Xd0TxbZA#ghhjmBHi8gJtsFgY4AkL8w))

You will first need to upgrade the "Code file" (aka bin file). Select the "Code file download mode" radio button, this will enable the related subsection, select the bin file using the "..." button on the right (you can download it from [here](https://mega.nz/folder/Xd0TxbZA#ghhjmBHi8gJtsFgY4AkL8w)) and press Download button at the bottom.
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230363083-95d0d300-5cbd-4854-84f7-23e0f3b8ef51.png" width=100% height=100%>
</p>

Then wait about 30 seconds for the device to finish the flashing (LED stops blinking).
Now select "Webpage directly download mode" radio button and select the folder containing the files (make sure you actually select the folder containing the files)
<p align="center">
  <img src="https://user-images.githubusercontent.com/4704172/230364904-5fae82f5-7584-4bec-bce9-7d0cfac03e1c.png" width=100% height=100%>
</p>

Hit Download again.
Wait 30 seconds. Restart. Done. (you can verify the new FW has been flashed by going to the "Device Settings" window and check under the "Device Info" section that the "Firmware version" matches the one you just flashed).

[WIP]
- Change device user & password ("Modify Key")
- Clear up confusion on Save/Load Default setting

________________________________________________________
Thanks to Gianfranco Falco who helped me with this.

**Comment by canton7** [comment]
Awesome, thank you!

@nathanmarlor Is it worth putting this up on the wiki do you think? Could you make it open to all (if it isn't already) -- I've never had that be a problem, and means that people can contribute

**Reply by Ndrinta** [reply]
I believe having a little firewall on the Wiki is a good thing, though...

**Reply by canton7** [reply]
I've got the documentation wikis open on a few of my projects, and I've never had an edit that I needed to roll back, in practice. 

On the flipside, if you restrict it you get a situation such as StealthChesnut's wiki, where there is a list of RS485 registers which is out of date, noone can edit it, and everyone just ends up maintaining their own little lists and sharing updates in a rather ad-hoc way.

**Reply by Ndrinta** [reply]
Yeah, fair enough. Not joking when I'm saying that that bothers me on a daily basis.

**Reply by nathanmarlor** [reply]
Consider it done, thanks @Ndrinta !

**Reply by nathanmarlor** [reply]
Done, done. https://github.com/nathanmarlor/foxess_modbus/wiki/Waveshare-RS485-to-ETH-(B)-Setup-Guide
